### DISCUSSION TITLE
Questions about `--stats-internal`

### DISCUSSION THREAD

**Discussion by hagozaebii** [cvc5/cvc5#11245] [Q&A] [answered]
In previous versions of CVC5, using the `--stats-internal` option would display solver statistics (e.g., sat::clauses_literals) even if the solver couldn't find a solution within the given time limit. However, I've noticed that in recent versions, this behavior seems to have changed.

I'm looking for a way to retrieve the "solver progress information up to the timeout" in the latest versions of CVC5, even when the solver doesn't successfully solve the problem within the allocated time. Specifically, I'm interested in statistics like the number of conflicts encountered, decisions made, and other relevant solver metrics.

Is there a current option or method to obtain these intermediate statistics when CVC5 times out? If the --stats-internal option no longer provides this functionality, are there alternative flags or approaches to access this information?
Any guidance on how to extract these solver statistics in timeout scenarios using the latest version of CVC5 would be greatly appreciated.

**Comment by hagozaebii** [comment]
I guess that cvc5 does not provide `--stats-internal` information when cvc5 does not finish solving as of version `1.2.0`.

**Comment by mpreiner** [comment, marked as answer]
If you use --tlimit-per the statistics will still be printed. If you use --tlimit the process is killed immediately without printing statistics.

**Reply by hagozaebii** [reply]
@mpreiner Thanks for your kindness!
